src/packages/cli/src/bin.ts

```diff
@@ -88,7 +91,33 @@ if (process.argv.length > 2) {
       debug('Environment variables not loaded (--schema was provided)')
     }
   }
-  // 2 - Check ./prisma directory for schema.prisma
+  // 2 - Check package.json for `prisma.schema` configuration
+  else if (
+    (pkgJsonPath = pkgUp.sync({ cwd: process.cwd() })) !== null &&
+    (pkgJsonContent = JSON.parse(fs.readFileSync(pkgJsonPath, 'utf-8'))) &&
+    pkgJsonContent.prisma?.schema !== undefined
+  ) {
```

Comment:
Not my best code ever 🙈, but given the current else-if structure there isn't a much better solution, except refactoring it into a small helper. I tried following the current code structure (raw if statements), I can refactor this if it's found too hairy.